gatewayconfig/bluetooth/characteristics/wifi_connect_characteristic.py

```diff
@@ -22,17 +24,26 @@ def __init__(self, service):
                 ["read", "write", "notify"], service)
         self.add_descriptor(WifiConnectDescriptor(self))
         self.add_descriptor(OpaqueStructureDescriptor(self))
-        self.wifi_status = ""
+        self.wifi_service = ""
+        self.wifi_password = ""
+
+    def wifi_connect(self, wifi_service, wifi_password):
+        if self.check_wifi_status() == "connected":
+            nmcli_custom.device.disconnect('wlan0')
+            logger.debug('Disconnected From Wifi')
+
+        nmcli_custom.device.wifi_connect(wifi_service, wifi_password)
 
     def WiFiConnectCallback(self):
         if self.notifying:
             logger.debug('Callback WiFi Connect')
             value = []
-            self.wifi_status = "timeout"
+            wifi_status = self.check_wifi_status()
 
-            for c in self.wifi_status:
+            for c in wifi_status:
```

Comment:
Since we are touching this, can we also use [this](https://github.com/NebraLtd/hm-config/blob/e9ca13990473e2a28179ac2ccbf85dd44c5680a8/gatewayconfig/helpers.py#L6) helper?

Reply:
Sure.